typescript/src/redemption.ts

```diff
@@ -264,3 +265,38 @@ export async function createRedemptionTransaction(
     transactionHex: transaction.toRaw().toString("hex"),
   }
 }
+
+/**
+ * Prepares the proof of a redemption transaction and submits it to the
+ * Bridge on-chain contract.
+ * @param transactionHash - Hash of the transaction being proven.
+ * @param mainUtxo - Recent main UTXO of the wallet as currently known on-chain.
+ * @param walletPubKeyHash - 20-byte public key hash of the wallet
+ * @param bridge - Interface to the Bridge on-chain contract.
+ * @param bitcoinClient - Bitcoin client used to interact with the network.
+ * @returns Empty promise.
+ */
+export async function proveRedemption(
+  transactionHash: string,
+  mainUtxo: UnspentTransactionOutput,
+  walletPubKeyHash: string,
+  bridge: Bridge,
+  bitcoinClient: BitcoinClient
+): Promise<void> {
+  const confirmations = await bridge.txProofDifficultyFactor()
+  const proof = await createTransactionProof(
+    transactionHash,
+    confirmations,
+    bitcoinClient
+  )
+  // TODO: instead of getting rawTransaction, use transaction part of proof and
+  // convert it to raw transaction.
```

Comment:
Can we put a note (as well in the same TODO in `proveDepositSweep`) explaining why we can't do it right now?

Reply:
Actually, I don't know why we cannot do that now. Can't we just use `rawTransaction` downloaded from from the `bitcoinClient`? Using the converter may result in fewer network connections though.

Reply:
This is because we don't have the right converter yet. We can use `rawTransaction` for now. My comment was about adding an explanation saying what should be done to consider this TODO as addressed.

Reply:
Done in d4b6523c143327cd52a483bc6d6e73764edf194a.